### PR TITLE
Synching the generator with the latest changes to the generated default protocol

### LINKED ISSUE #520: Describe custom types in protocol specification using json schema for automatic generation
**Is your feature request related to a problem? Please describe.**
Currently custom types cannot be formally described in the protocol specification yaml file.

**Describe the solution you'd like**
@marcofavorito suggests custom types could be described in the yaml file using something similar to json

### PR DESCRIPTION
**Pull request by 5A11**
## Proposed changes

Synching the generator with the latest changes to the generated default protocol
Cleanups and styling

Fix #520

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/protocols/default/custom_types.py [hunk 1]
```diff
@@ -35,12 +35,29 @@ class ErrorCode(Enum):
     def encode(
         cls, performative, error_code_from_message: "ErrorCode",
     ):
-        """Encode an instance of this class into its protobuf object."""
+        """
+        Encode an instance of this class into the protocol buffer object.
+
+        'performative' argument is the protocol buffer object containing a content whose type is this class.
```
Comment (DavidMinarsch) [inline aea/protocols/default/custom_types.py]: This is a repetition of the parameter definition, I'd remove line 41

File: aea/protocols/default/custom_types.py [hunk 2]
```diff
@@ -35,12 +35,29 @@ class ErrorCode(Enum):
...
         performative.error_code.error_code = error_code_from_message.value
         return performative
 
     @classmethod
     def decode(cls, error_code_from_pb2) -> "ErrorCode":
-        """Decode an instance of this class that has been serialised."""
+        """
+        Decode a protocol buffer object that corresponds with this class into an instance of this class.
+
+        'error_code_from_pb2' argument is the protocol buffer content object whose type corresponds with this class.
```
Comment (DavidMinarsch) [inline aea/protocols/default/custom_types.py]: See above, make it more succinct please as there is repetition between this and the parameter definition

File: aea/protocols/default/custom_types.py (hunk 1 above)
Comment (5A11) [reply aea/protocols/default/custom_types.py]: Done

File: aea/protocols/default/custom_types.py (hunk 2 above)
Comment (5A11) [reply aea/protocols/default/custom_types.py]: Done

**Comment by DavidMinarsch** [comment]
> In message.py, do you refer to the as CustomErrorCode bit in the import statement that we talked about in the morning?

Yes

> Do you want the generator to follow that style?

Yes

:)

**Pull request merged**